src/kvs/server.cpp

```diff
@@ -63,6 +63,14 @@ void run(unsigned thread_id, Address public_ip, Address private_ip,
 
   // prepare the zmq context
   zmq::context_t context(1);
+
+  auto res = context.setctxopt(ZMQ_MAX_SOCKETS, kMaxSocketNumber);
+  if (res == 0) {
+    log->info("Successfully set max socket number to {}", kMaxSocketNumber);
+  } else {
+    log->error("E: socket error number {} ({})", errno, zmq_strerror(errno));
```

Comment:
Is this a catastrophic failure? Or is it fine if this fails?

Reply:
not a catastrophic failure. If it fails it will keep the default limit which is 1024.